File: ProyectoAccesoDatos-AJA/src/main/java/com/example/proyecto/domain/Valoracion.java
package com.example.proyecto.domain;

import jakarta.persistence.*;
import java.time.LocalDate;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "idValoracion")
@Entity
@Table(name = "Valoracion")
public class Valoracion {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id_valoracion")
    private Integer idValoracion;
    @OneToOne
    @JoinColumn(name="id_cita")
    private Cita cita;
    @Column(name="puntuacion")
    private Integer puntuacion;
    @Column(name="comentario")
    private String comentario;
    @Column(name="fecha_valoracion")
    private LocalDate fechaValoracion;

    public Integer getIdValoracion(){return idValoracion;}
    public void setIdValoracion(Integer idValoracion){this.idValoracion=idValoracion;}
    public Cita getCita(){return cita;}
    public void setCita(Cita cita){this.cita=cita;}
    public Integer getPuntuacion(){return puntuacion;}
    public void setPuntuacion(Integer puntuacion){this.puntuacion=puntuacion;}
    public String getComentario(){return comentario;}
    public void setComentario(String comentario){this.comentario=comentario;}
    public LocalDate getFechaValoracion(){return fechaValoracion;}
    public void setFechaValoracion(LocalDate fechaValoracion){this.fechaValoracion=fechaValoracion;}
}

File: ProyectoFlutter/lib/models/valoracion.dart
class Valoracion {
  final int? idValoracion;
  final int puntuacion;
  final String comentario;
  final int idCita;
  final String? fechaValoracion;

  Valoracion({
    this.idValoracion,
    required this.puntuacion,
    required this.comentario,
    required this.idCita,
    this.fechaValoracion,
  });

  factory Valoracion.fromJson(Map<String, dynamic> json) {
    return Valoracion(
      idValoracion: json["idValoracion"] ?? json["id_valoracion"],
      puntuacion: json["puntuacion"] ?? 0,
      comentario: (json["comentario"] ?? "").toString(),
      idCita: _parseCitaId(json["cita"]) ?? json["id_cita"] ?? 0,
      fechaValoracion: json["fechaValoracion"] ?? json["fecha_valoracion"],
    );
  }

  static int? _parseCitaId(dynamic data) {
    if (data == null) return null;
    if (data is int) return data;
    return data["idCita"] ?? data["id_cita"];
  }

  Map<String, dynamic> toJson() {
    return {
      if (idValoracion != null) "idValoracion": idValoracion,
      "puntuacion": puntuacion,
      "comentario": comentario,
      "cita": {"idCita": idCita},
      if (fechaValoracion != null) "fechaValoracion": fechaValoracion,
    };
  }
}
